network/manager.go

```diff
@@ -324,9 +309,54 @@ func (m *Manager) watchNetworks() {
 	}
 }
 
-func (m *Manager) Run() {
+func (m *Manager) addInitialNetworks(ctx context.Context) bool {
+	if m.isMultiNetwork() {
+		// Get list of existing networks
+		result, err := m.sm.WatchNetworks(ctx, nil)
+		if err != nil {
+			return false
+		}
+
+		for _, n := range result.Snapshot {
+			if m.isNetAllowed(n) {
+				m.networks[n] = NewNetwork(ctx, m.sm, m.bm, n, m.ipMasq)
+			}
+		}
+	} else {
+		if _, err := m.sm.GetNetworkConfig(ctx, ""); err != nil {
```

Comment:
I added the GetNetworkConfig() for consistency with the multinetwork case.  Previously, this would have simply passed because NewNetwork() doesn't require any kind of etcd/server connectivity.  So you would then get to Network.Run() and hit that retry loop, I suppose.  This way, both codepaths ask for the etcd/server data at the same time.

I'm happy to remove this new bit if you like, and leave it up to Network.Run()?  Just thought some consistency of behavior might be useful.